pkg/eventemitter/eventemitter.go

```diff
@@ -193,10 +213,27 @@ func (e *EventEmitter) createEventHandlers(ctx context.Context, cloudEventSource
 			h.CloseHandler()
 		}
 		e.eventHandlersCache[eventHandlerKey] = eventHandler
+		return
 	}
 
 	// Create EventFilter from CloudEventSource
 	e.eventFilterCache[key] = NewEventFilter(cloudEventSource.Spec.EventSubscription.IncludedEventTypes, cloudEventSource.Spec.EventSubscription.ExcludedEventTypes)
```

Comment:
We need to move this filter before the HTTP destination return or it won't apply to it